src/MCControl.cpp

```diff
@@ -128,6 +134,78 @@ MCControl::MCControl(RTC::Manager* manager)
           "RobotModule contains a JointSensor at {} but it was not found in the refJointOrder.", js.joint());
     }
   }
+
+  controller.controller().datastore().make_call(
+      controller.robot().name() + "::SetPDGains",
+      [this](const std::vector<double> & p_vec, const std::vector<double> & d_vec) {
+        const auto & rjo = controller.robot().refJointOrder();
+        if(p_vec.size() != rjo.size())
+        {
+          return false;
+        }
+        if(d_vec.size() != rjo.size())
+        {
+          return false;
+        }
+
+#ifdef USE_IOB
+        // if on real robot
+        open_iob();
+        for(unsigned int i = 0; i < rjo.size(); i++)
+        {
+          write_pgain(i, p_vec[i]);
+          write_dgain(i, d_vec[i]);
+        }
+        close_iob();
+#else
+        // if not on real robot
+        m_pgainsOut.data.length(rjo.size());
+        m_dgainsOut.data.length(rjo.size());
+        for(unsigned int i = 0; i < rjo.size(); i++)
+        {
+          m_pgainsOut.data[i] = p_vec[i];
+          m_dgainsOut.data[i] = d_vec[i];
+        }
+
+        coil::TimeValue coiltm(coil::gettimeofday());
+        RTC::Time tm;
+        tm.sec = static_cast<CORBA::ULong>(coiltm.sec());
+        tm.nsec = static_cast<CORBA::ULong>(coiltm.usec()) * 1000;
+        m_pgainsOut.tm = tm;
+        m_dgainsOut.tm = tm;
+        m_pgainsOutOut.write();
+        m_dgainsOutOut.write();
+#endif
+        return true;
```

Comment:
Would be cleaner as a method that is called by the datastore rather than a long lambda

Reply:
Done.